starlite/middleware/base.py

```diff
@@ -8,20 +8,24 @@
 
 @runtime_checkable
 class MiddlewareProtocol(Protocol):  # pragma: no cover
-    def __init__(self, app: "ASGIApp", **kwargs: Dict[str, Any]):
-        """The MiddlewareProtocol is a PEP 544 Protocol that species the
-        requirements for an ASGI middleware.
+    def __init__(self, app: "ASGIApp", **kwargs: Any):
```

Comment:
what does this fix?

Reply:
`**kwargs` aren't a `dict`, e.g., I can't call that method with `MyMiddleware(app, {"k": "v"})`. The key value of `**kwargs` is always a `str` so that is inferred and only the value type needs to be specified.

Ref: https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values

Reply:
it is a dict - but destructured, you can call `MyMiddleware(app, **{"k": "v"})`. Furthermore, you can call all the dict methods on kwargs.

Reply:
It isn't redundant, it is correct.

```python
from typing import Any


def fn_correct(**kwargs: Any) -> None:
    reveal_type(kwargs)  # main.py:5: note: Revealed type is "builtins.dict[builtins.str, Any]"
 


def fn_incorrect(**kwargs: dict[str, Any]) -> None:
    reveal_type(kwargs)  # main.py:9: note: Revealed type is "builtins.dict[builtins.str, builtins.dict[builtins.str, Any]]"

```

Reply:
ok, then can u replace all other occurances while at it?